src/mission.cpp

```diff
@@ -416,6 +417,20 @@ void mission::wrap_up()
         case MGOAL_FIND_ANY_ITEM:
             player_character.remove_mission_items( uid );
             break;
+        case MGOAL_FIND_MONSTER: {
+            Creature *found_creature = g->get_creature_if( [&]( const Creature & critter ) {
+                const monster *const mon_ptr = dynamic_cast<const monster *>( &critter );
+                return mon_ptr && mon_ptr->mission_ids.count( uid );
+            } );
+            if( found_creature != nullptr ) {
+                monster *found_monster = found_creature->as_monster();
+                if( found_monster != nullptr ) {
+                    found_monster->mission_ids.erase( uid );
+                    found_monster->remove_effect( effect_pet );
+                    found_monster->friendly = 0;
```

Comment:
You can probably make the pet friendly to the quest giver by assigning their faction to the pet. Try something like this:
```suggestion
                    found_monster->friendly = 0;
                    const npc *n = g->find_npc( npc_id );
                    if( n ) {
                        found_monster->faction = n->get_faction()->mon_faction;
                    }
```

Reply:
I added this code in my local branch,  I was able to observe the change in the save file (the faction changes to "human"), but I did not observe any change in the behavior of the animals; the dog still stood around in the lobby of the Refugee Center, and the cat still ran away almost immediately.  

After further investigating the issue seems to be that the cat base monster "cat_mon", has the "fear_triggers" of "PLAYER_CLOSE", which results in the cat losing morale and running away.  I tried increasing the cat's morale to 100 which causes the cat to hang nearby for a few ticks, but it still runs away very quickly.  I then tried removing the "PLAYER_CLOSE" "fear_triggers" and the cat no longer ran away.

I then modified the code provided by @dseguin so that the monster is added to the character's faction "player".  This results in the cat showing as friendly (it has a green dot on the monster icon, and is listed as friendly in the V panel), but the cat still flees from the character.

I think this probably could be resolved by adding a new set of "mon_cat" monster types that do not have the "PLAYER_CLOSE" "fear_triggers", but that seems a bit error prone if someone adds a new "mon_cat" subtype and forgets to add the corresponding type that does have the "PLAYER_CLOSE" "fear_triggers".

I will also spend some time debugging through the monmove.cpp monster::plan code to see if I can see any other options that could be used to change this outcome.

Reply:
After reading and debugging and logging the monmove.cpp monster::plan code I am unwilling to change it due to the risk of unintended consequences.

I then pursued the idea of creating a new type of cat.  This actually worked out better than I had hoped.  I was able to add one new cat entry (mon_cat_isherwood), that does not have the "PLAYER_CLOSE" "fear_triggers" and I gave it a higher morale.  The combination of these results in the cat hanging around the Isherwood house after it has been returned to Claire, and it has the side effect of addressing @PatrikLundell's concern about the "catchability" of the cats.  The player can now get closer to the cats before the cats start fleeing, and (if the player runs) a cat can now be caught without needing to teleport.